Lib/test/test_zipimport.py

```diff
@@ -669,6 +669,22 @@ def testBytesPath(self):
 class CompressedZipImportTestCase(UncompressedZipImportTestCase):
     compression = ZIP_DEFLATED
 
+    @support.cpython_only
+    def test_issue31602(self):
+        # There shouldn't be an assertion failure in zipimporter.get_source()
+        # in case of a bad zlib.decompress().
+        import zlib
+        def bad_decompress(*args):
+            return None
+        with ZipFile(TEMP_ZIP, 'w') as zip_file:
+            self.addCleanup(support.unlink, TEMP_ZIP)
+            name = 'bar.py'
```

Comment:
Why not inline these values? For clarity use a valid Python statement instead of 'spam'.

Reply:
No idea why not. I just copied the style of all other tests in this file.